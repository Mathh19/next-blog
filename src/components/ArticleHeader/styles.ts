import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

export const Wrapper = styled.header`
  ${({ theme }) => css`
    padding-bottom: ${theme.spacings.xlarge};
    margin-bottom: ${theme.spacings.xlarge};
    border-bottom: 0.1rem solid ${theme.colors.mediumGray};

    ${HeadingStyles} {
      margin: 0;
      margin-bottom: ${theme.spacings.medium};
    }
  `}
`;

export const Excerpt = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacings.medium} 0;
    font-size: ${theme.font.sizes.medium};
  `}
`;

export const CoverContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    justify-content: center;
    margin-bottom: ${theme.spacings.medium};
  `}
`;

export const Cover = styled.img`
  ${css`
    max-width: 100%;
    display: block;
  `}
`;
