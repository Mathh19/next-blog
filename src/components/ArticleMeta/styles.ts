import styled, { css } from 'styled-components';

export const Wrapper = styled.article`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.darkGray};
    font-style: italic;

    .categories span::after {
      content: ', ';
    }
    .categories span:last-child::after {
      content: '';
    }

    a {
      color: ${theme.colors.secondary};
      text-decoration: none;
      transition: all 300ms ease-in-out;

      &:hover {
        filter: brightness(50%);
      }
    }
  `}
`;
