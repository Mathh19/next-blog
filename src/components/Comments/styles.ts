import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    max-width: ${theme.sizes.content};
    width: 100%;
    margin: ${theme.spacings.large} auto;
    padding: 0 ${theme.spacings.large};
  `}
`;

export const NoComments = styled.div`
  ${({ theme }) => css`
    text-align: center;
    color: ${theme.colors.secondary};
    font-size: ${theme.font.sizes.medium};
  `}
`;
