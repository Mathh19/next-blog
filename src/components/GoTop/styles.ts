import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    background-color: #082032;
    color: ${theme.colors.secondary};
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;
    bottom: 2rem;
    right: 2rem;
    z-index: 6;
    opacity: 0.7;
    border-radius: 50%;
    border: 0.2rem solid ${theme.colors.secondary};
    &:hover {
      transition: 500ms ease-in-out all;
      background-color: #12324a;
      opacity: unset;
    }
  `}
`;
