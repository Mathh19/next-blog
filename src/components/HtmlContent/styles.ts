import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    font-size: 1.8rem;
    line-height: 1.5;

    p {
      margin: ${theme.spacings.medium} 0;
    }

    a {
      color: ${theme.colors.secondary};
    }

    a:visited,
    a:link {
      text-decoration: none;
      transition: all 300ms ease-in-out;
    }

    a:hover {
      filter: brightness(50%);
    }

    pre {
      background-color: #2C394B;
      padding: ${theme.spacings.medium};
      font-family: monospace;
      color: #B2C8DF;
      margin: ${theme.spacings.xlarge} 0;
      width: 100%;
      font-size: ${theme.font.sizes.small};
      overflow-x: auto;
    }

    pre code {
      background: inherit;
      color: inherit;
    }

    code {
      font-family: monospace;
      color: #B2C8DF;
      font-size: ${theme.font.sizes.small};
      background: #2C394B;
      padding: 0.2rem;
      margin: 0.2rem;
      border-radius: 5px;
    }

    img {
      max-width: 100%;
    }

    .image {
      background: ${theme.colors.mediumGray};
      line-height: 0;
      margin: ${theme.spacings.medium} 0;
    }

    .image figcaption {
      font-size: ${theme.font.sizes.small};
      color: #000000;
      padding: ${theme.spacings.small};
      text-align: center;
      line-height: 1.3;
    }

    .image-style-side {
      float: right;
      max-width: 50%;
      margin: ${theme.spacings.medium};
    }

    hr {
      border: none;
      border-bottom: 0.1rem solid ${theme.colors.mediumGray};
    }

    ul, ol {
      margin: ${theme.spacings.medium};
    }

    .table {
      width: 100%;
      overflow: hidden;
      overflow-x: auto;
    }

    table {
        width: 100%;
        border-collapse: collapse;
        margin: ${theme.spacings.medium} 0;
    }

    table td, table th {
      padding: ${theme.spacings.small};
      border: 0.1rem solid ${theme.colors.mediumGray};
    }

    blockquote {
      border-left: 0.5rem solid ${theme.colors.secondary};
      color: ${theme.colors.darkGray};
      filter: brightness(80%);
      padding-left: ${theme.spacings.medium};
      font-style: italic;
      margin: ${theme.spacings.medium};
    }

    @media ${theme.media.lteMedium} {
      font-size: 2rem;

      .image-style-side {
      float: none;
      max-width: 100%;
      margin: 0;
    }
    }

  `}
`;
