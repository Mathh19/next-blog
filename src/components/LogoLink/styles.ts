import styled from 'styled-components';

export const Container = styled.a`
    display: flex;
    align-items: center;
    text-decoration: none;
    color: inherit;

    > img {
      width: 150px;
      height: 150px;
      border-radius: 50%;
    }
`;
