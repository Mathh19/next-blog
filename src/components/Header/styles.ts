import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

export const Wrapper = styled.header`
  ${({ theme }) => css`
  margin: 0 auto;
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  max-width: ${theme.sizes.max};
  padding-bottom: ${theme.spacings.xhuge};
  font-size: ${theme.font.sizes.small};
  color: ${theme.colors.darkText};

  ${HeadingStyles} {
    margin: 0 0 0.6rem;
  }
  `}

  img:hover {
    transition: all ease-in-out 300ms;
    filter: brightness(70%);
  }
`;

export const Content = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-flow: column wrap;
    margin-left: ${theme.spacings.large};
    max-width: 48rem;
  `}
`;
