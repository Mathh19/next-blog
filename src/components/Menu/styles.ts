import styled, { css, DefaultTheme } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

type MenuBehaviourProps = {
  menuVisible: boolean;
};

const wrapperChanger = (menuVisible: MenuBehaviourProps['menuVisible']) => css`
  left: ${menuVisible ? '0' : '-36rem'};
  overflow-y: ${menuVisible ? 'auto' : 'hidden'};
`;

export const Wrapper = styled.div<MenuBehaviourProps>`
  ${({ theme, menuVisible }) => css`
    background: ${theme.colors.bgMenu};
    padding: ${theme.spacings.large};
    display: flex;
    position: fixed;
    z-index: 1;
    width: 100%;
    max-width: 32rem;
    height: 100vh;
    top: 0;
    left: 0rem;
    transition: all 400ms ease-in-out;
    overflow-y: auto;
    box-shadow: 5px 0 10px -2px ${theme.colors.black};
    scrollbar-color: ${theme.colors.secondary} ${theme.colors.primary};
    scrollbar-width: thin;
    ${wrapperChanger(menuVisible)}


    ::-webkit-scrollbar {
      width: 10px;
    }

    ::-webkit-scrollbar-track {
      background: ${theme.colors.primary};
    }

    ::-webkit-scrollbar-thumb {
      background: ${theme.colors.secondary};
      border-radius: 10px;
    }

    ::-webkit-scrollbar-thumb:hover {
      background: #B21031;
    }
  `}
`;

export const Nav = styled.nav`
    margin: auto;
    width: 100%;
`;

export const Logo = styled.div`
  ${({ theme }) => css`
    ${HeadingStyles} {
      display: flex;
      justify-content: center;
      margin: 0;
      margin-bottom: ${theme.spacings.xxlarge};
    }

    img {
      border: 0.5rem solid ${theme.colors.secondary};
    }
  `}
`;

const buttonChanger = (
  menuVisible: MenuBehaviourProps['menuVisible'],
  theme: DefaultTheme,
) => css`
  left: ${menuVisible ? '26rem' : '1rem'};
  color: ${theme.colors.secondary};

  @media ((max-width: 300px)) {
    left: ${menuVisible ? '23rem' : '1rem'};
  }
`;

export const OpenClose = styled.a<MenuBehaviourProps>`
  ${({ theme, menuVisible }) => css`
    position: fixed;
    top: ${theme.spacings.medium};
    color: ${theme.colors.white};
    background: #082032;
    z-index: 2;
    width: 3rem;
    height: 3rem;
    left: 26rem;
    border-radius: 4px;
    transition: all 400ms ease-in-out;
    ${buttonChanger(menuVisible, theme)}
  `}
`;
