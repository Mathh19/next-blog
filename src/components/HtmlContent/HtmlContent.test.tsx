import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { HtmlContent } from './index';

describe('<HtmlContent />', () => {
  test('it should render a markdown', () => {
    renderTheme(<HtmlContent html="### Children" />);
    expect(screen.getByText('Children')).toBeInTheDocument();
  });

  test('it should match snapshot', () => {
    const { container } = renderTheme(<HtmlContent html="### Children" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
