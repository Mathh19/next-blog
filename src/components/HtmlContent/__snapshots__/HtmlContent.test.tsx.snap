// Vitest Snapshot v1

exports[`<HtmlContent /> > it should match snapshot 1`] = `
.c0 {
  font-size: 1.8rem;
  line-height: 1.5;
}

.c0 p {
  margin: 2.4rem 0;
}

.c0 a {
  color: #E94560;
}

.c0 a:visited,
.c0 a:link {
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c0 a:hover {
  -webkit-filter: brightness(50%);
  filter: brightness(50%);
}

.c0 pre {
  background-color: #2C394B;
  padding: 2.4rem;
  font-family: monospace;
  color: #B2C8DF;
  margin: 4.0rem 0;
  width: 100%;
  font-size: 1.6rem;
  overflow-x: auto;
}

.c0 pre code {
  background: inherit;
  color: inherit;
}

.c0 code {
  font-family: monospace;
  color: #B2C8DF;
  font-size: 1.6rem;
  background: #2C394B;
  padding: 0.2rem;
  margin: 0.2rem;
  border-radius: 5px;
}

.c0 img {
  max-width: 100%;
}

.c0 .image {
  background: #DDDDDD;
  line-height: 0;
  margin: 2.4rem 0;
}

.c0 .image figcaption {
  font-size: 1.6rem;
  padding: 1.6rem;
  text-align: center;
  line-height: 1.3;
}

.c0 .image-style-side {
  float: right;
  max-width: 50%;
  margin: 2.4rem;
}

.c0 hr {
  border: none;
  border-bottom: 0.1rem solid #DDDDDD;
}

.c0 ul,
.c0 ol {
  margin: 2.4rem;
}

.c0 .table {
  width: 100%;
  overflow: hidden;
  overflow-x: auto;
}

.c0 table {
  width: 100%;
  border-collapse: collapse;
  margin: 2.4rem 0;
}

.c0 table td,
.c0 table th {
  padding: 1.6rem;
  border: 0.1rem solid #DDDDDD;
}

.c0 blockquote {
  border-left: 0.5rem solid #E94560;
  color: #AAAAAA;
  -webkit-filter: brightness(80%);
  filter: brightness(80%);
  padding-left: 2.4rem;
  font-style: italic;
  margin: 2.4rem;
}

@media (max-width:768px) {
  .c0 {
    font-size: 2rem;
  }

  .c0 .image-style-side {
    float: none;
    max-width: 100%;
    margin: 0;
  }
}

<p
  class="c0"
>
  <b>
    Children
  </b>
</p>
`;
